test/crl-shard-checker/crl-shard-checker.go

```diff
@@ -0,0 +1,154 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"time"
+
+	"github.com/letsencrypt/boulder/cmd"
+)
+
+// startTime is used as a universal starting point against which other times
+// can be compared, because the Go zero time is too far in the past for math
+// to work correctly. The significance of 2009-11-10 23:00:00 is left as an
+// exercise to the reader.
+var startTime = time.Date(2009, time.November, 10, 23, 00, 00, 0, time.UTC)
+
+type crlUpdater struct {
+	numShards         int
+	lookbackPeriod    time.Duration
+	lookforwardPeriod time.Duration
+}
+
+// getShardBoundaries computes the start (inclusive) and end (exclusive) times
+// for a given integer-indexed CRL shard. The idea here is that shards should be
+// stable. Picture a timeline, divided into chunks. Number those chunks from 0
+// to cu.numShards, then repeat the cycle when you run out of numbers:
+//
+//	   chunk:  5     0     1     2     3     4     5     0     1     2     3
+//	...-----|-----|-----|-----|-----|-----|-----|-----|-----|-----|-----|-----...
+//	                         ^  ^-atTime                         ^
+//	   atTime-lookbackPeriod-┘          atTime+lookforwardPeriod-┘
+//
+// The width of each chunk is determined by dividing the total time window we
+// care about (lookbackPeriod+lookforwardPeriod) by the number of shards we
+// want (numShards).
+//
+// Even as "now" (atTime) moves forward, and the total window of expiration
+// times that we care about moves forward, the boundaries of each chunk remain
+// stable:
+//
+//	   chunk:  5     0     1     2     3     4     5     0     1     2     3
+//	...-----|-----|-----|-----|-----|-----|-----|-----|-----|-----|-----|-----...
+//	                                 ^  ^-atTime                         ^
+//	           atTime-lookbackPeriod-┘          atTime+lookforwardPeriod-┘
```

Comment:
Do we really want to duplicate this whole comment? Seems likely to get out of sync and sow doubt about the correctness of both comments.

Reply:
Ahahah sorry, no, this file isn't even supposed to be in this PR at all! I'm not even sure when it got added to the commit. This was just my own testbed for verifying how shard boundaries were changing.